packages/flutter/lib/src/material/contained_button.dart

```diff
@@ -0,0 +1,425 @@
+  /// * `overlayColor`
+  ///   * hovered - Theme.colorScheme.onPrimary(0.08)
+  ///   * focused or pressed - Theme.colorScheme.onPrimary(0.24)
+  /// * `shadowColor` - Colors.black
+  /// * `elevation`
+  ///   * disabled - 0
+  ///   * hovered or focused - 2
+  ///   * pressed - 6
+  /// * `padding`
+  ///   * textScaleFactor <= 1 - horizontal(16)
+  ///   * `1 < textScaleFactor <= 2` - lerp(horizontal(16), horizontal(8))
+  ///   * `2 < textScaleFactor <= 3` - lerp(horizontal(8), horizontal(4))
+  ///   * `3 < textScaleFactor` - horizontal(4)
+  /// * `minimumSize` - Size(64, 36)
+  /// * `side` - BorderSide.none
+  /// * `shape` - RoundedRectangleBorder(borderRadius: BorderRadius.circular(4))
+  /// * `mouseCursor`
+  ///   * disabled - SystemMouseCursors.forbidden
+  ///   * others - SystemMouseCursors.click
+  /// * `visualDensity` - theme.visualDensity
+  /// * `tapTargetSize` - theme.materialTapTargetSize
+  /// * `animationDuration` - kThemeChangeDuration
+  /// * `enableFeedback` - true
+  ///
+  /// The default padding values for the [ContainedButton.icon] factory are slightly different:
+  ///
+  /// * `padding`
+  ///   * `textScaleFactor <= 1` - start(12) end(16)
+  ///   * `1 < textScaleFactor <= 2` - lerp(start(12) end(16), horizontal(8))
+  ///   * `2 < textScaleFactor <= 3` - lerp(horizontal(8), horizontal(4))
+  ///   * `3 < textScaleFactor` - horizontal(4)
+
+  @override
+  ButtonStyle defaultStyleOf(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final ColorScheme colorScheme = theme.colorScheme;
+
+    final EdgeInsetsGeometry scaledPadding = ButtonStyleButton.scaledPadding(
+      const EdgeInsets.symmetric(horizontal: 16),
+      const EdgeInsets.symmetric(horizontal: 8),
+      const EdgeInsets.symmetric(horizontal: 4),
+      MediaQuery.of(context, nullOk: true)?.textScaleFactor ?? 1,
+    );
+
+    return styleFrom(
+      primary: colorScheme.primary,
+      onPrimary: colorScheme.onPrimary,
+      onSurface: colorScheme.onSurface,
+      shadowColor: const Color(0xFF000000),
+      elevation: 2,
+      textStyle: theme.textTheme.button,
+      padding: scaledPadding,
+      minimumSize: const Size(64, 36),
+      side: BorderSide.none,
+      shape: const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(4))),
+      enabledMouseCursor: SystemMouseCursors.click,
+      disabledMouseCursor: SystemMouseCursors.forbidden,
+      visualDensity: theme.visualDensity,
+      tapTargetSize: theme.materialTapTargetSize,
+      animationDuration: kThemeChangeDuration,
+      enableFeedback: true,
+    );
+  }
+
+  /// Returns the [ContainedButtonThemeData.style] of the closest
+  /// [ContainedButtonTheme] ancestor.
+  @override
+  ButtonStyle themeStyleOf(BuildContext context) {
+    return ContainedButtonTheme.of(context)?.style;
+  }
+}
+
+@immutable
+class _ContainedButtonDefaultBackground extends MaterialStateProperty<Color>  with Diagnosticable {
```

Comment:
nit: double space here and the one below

```suggestion
class _ContainedButtonDefaultBackground extends MaterialStateProperty<Color> with Diagnosticable {
```

Reply:
Done